app/controllers/login.rb

```diff
@@ -0,0 +1,38 @@
+get '/login' do
+  erb :'users/login'
+end
+
+post '/login' do
+  if @user = User.authenticate(params[:username],params[:password])
+    session[:user_id] = @user.id
+    if request.xhr?
+      #TODO>>>>>>>
+    else
+      redirect '/'
+    end
+  else
+    @error_messages = ["Invalid username and/or password"]
+    erb :'users/login'
+  end
+end
+
+get '/logout' do
```

Comment:
since we are clearing the session here, the more RESTful convention is to make a `delete` request instead of a `get`